### DISCUSSION TITLE
Why .Length doesn't work?

### DISCUSSION THREAD

**Discussion by sea0920** [sebastienros/jint#1296] [Q&A]
The code below works fine if I call .length instead of .Length.
Why does this happen? Did C# string[] already turn into javascript array before calling .Length?

```
class Program
    {
        public class Foo
        {
            public string[] MyStrings { get; set; } = { "abc", "def" };
        }

        static void Main(string[] args)
        {
            var engine = new Engine();

            var result = engine
            .SetValue("Foo", typeof(Foo))
            .Evaluate("new Foo().MyStrings.Length") // Changing Length to length works.
            .ToString();

            Console.WriteLine(result);
        }
    }
```

**Comment by lahma** [comment]
I think in that case Jint does automatic conversion on the fly as `ArrayInstance` so it won't have any non-JS-properties. If you don't want for that to happen, you can configure `MemberAccessor` in interop options to return something else, like ObjectWrapper, but then you will lose the array behavior like `find` etc.

**Comment by dretax** [comment]
> The code below works fine if I call .length instead of .Length. Why does this happen? Did C# string[] already turn into javascript array before calling .Length?
> 
> ```
> class Program
>     {
>         public class Foo
>         {
>             public string[] MyStrings { get; set; } = { "abc", "def" };
>         }
> 
>         static void Main(string[] args)
>         {
>             var engine = new Engine();
> 
>             var result = engine
>             .SetValue("Foo", typeof(Foo))
>             .Evaluate("new Foo().MyStrings.Length") // Changing Length to length works.
>             .ToString();
> 
>             Console.WriteLine(result);
>         }
>     }
> ```
```
Engine.Execute(@"
                    Object.defineProperty(Array.prototype, 'Length', {
                        get: function () { return this.length; },
                        configurable: true
                    });

                    Object.defineProperty(String.prototype, 'Length', {
                        get: function () { return this.length; },
                        configurable: true
                    });

                    Object.defineProperty(Object.prototype, 'ToString', {
                        value: function () { return '' + this; },
                        writable: true,
                        configurable: true
                    });
                ");```
